### PR TITLE
Cardano Primitives

### PR DESCRIPTION
**Pull request by Sam-Jeston**
To improve the flexibility of this SDK, we abstract the cryptographic primitives available through the rust wasm binding behind an interface. This allows us to use other modules for these primitives as they become available.

So consequences of this refactor:
- Upgrade `cardano-wallet` dependency to the most recent version. This changed some interfaces, and the transaction.balance.is_negative behaviour has changed as well. I have not looked into this further as I think its outside our scope. Possibly we just require that inputs - outputs is a positive number, and let the network determine if the fee is appropriate. This is the web3 way
- Relevant top level interfaces now take `Cardano` as an optional parameter. These are currently defaulted to `RustCardano`

### CODE REVIEW

File: src/lib/Ledger/index.spec.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import { LedgerKeyManager } from '.'
-import { KeyManager } from '../KeyManager'
+import { KeyManager } from '../../KeyManager/KeyManager'
```
Comment (rhyslbw) [inline src/lib/Ledger/index.spec.ts]: As a general rule I think `index.ts` for `KeyManager` within a directory named `KeyManager` is a good style, or we could decide to go with exporting the module as a default from the index. Either way, let's make it consistent

File: src/index.ts [hunk 2]
```diff
@@ -1,18 +1,35 @@
-import Transaction from './Transaction'
+import Transaction, { TransactionInput, TransactionOutput } from './Transaction'
 import Wallet from './Wallet'
 import { Provider, HttpProvider } from './Provider'
 import * as Utils from './Utils'
-import { InMemoryKeyManager } from './KeyManager'
-
-export { Transaction, InMemoryKeyManager, Utils }
+import { InMemoryKeyManager, LedgerKeyManager, RustCardano } from './lib'
```
Comment (rhyslbw) [inline src/index.ts]: 👍 nice and clean

File: src/lib/Ledger/index.spec.ts (hunk 1 above)
Comment (Sam-Jeston) [reply src/lib/Ledger/index.spec.ts]: Ahhh that was just the editor

Comment (Sam-Jeston) [reply src/lib/Ledger/index.spec.ts]: > As a general rule I think index.ts for KeyManager within a directory named KeyManager is a good style
100%

**Pull request merged**